src/allotropy/parsers/utils/values.py

```diff
@@ -256,3 +260,31 @@ def get_attrib_from_xml(
     except KeyError as e:
         msg = f"Unable to find '{attrib_name}' in {xml_element.attrib}"
         raise AllotropeConversionError(msg) from e
+
+
+def remove_none_fields_from_data_class(
```

Comment:
Can we get some unit tests for this?

Reply:
@nathan-stender I added tests but had to ignore the incompatible type mypy errors when using `Optional[str]` since `make_dataclass` expected `type` not `object`

Reply:
No worries! 

I _think_ you could get around this by just defining the dataclasses outside of the test function.

But also, I don't really care about type ignores in a unit test :D